File: GYTableViewController/GYTableViewController/Demo/CreateTableDemo/CreateTableViewController.m
//
//  CreateTableController.m
//  GYTableViewController
//
//  Created by gaoyang on 2019/2/7.
//  Copyright © 2019年 高扬. All rights reserved.
//

#import "CreateTableViewController.h"
#import "PopupListView.h"
#import "Mock.h"

@interface CreateTableViewController ()

@property (nonatomic, strong) PopupListView *popupListView;

@property (nonatomic, strong) UIButton *startButton;

@end

@implementation CreateTableViewController

#pragma mark 主角是PopupListView中创建的自定义TableView实例
- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor whiteColor];
    
    self.startButton.frame = CGRectMake(0, 0, self.view.width, 50);
    
    [self.popupListView show];
}

- (void)p_tapStartButton {
    [self.popupListView show];
}

#pragma mark 懒加载添加视图
- (PopupListView *)popupListView {
    if (!_popupListView) {
        _popupListView = [[PopupListView alloc] init];
        _popupListView.title = @"世界上最牛B的语言";
        _popupListView.dataArray = Mock.popupOptions;
    }
    return _popupListView;
}


- (UIButton *)startButton {
    if (!_startButton) {
        _startButton = [UIButton buttonWithType:UIButtonTypeSystem];
        _startButton.backgroundColor = TVStyle.colorPrimaryDishes;
        [_startButton setTitle:@"弹  出" forState:UIControlStateNormal];
        [_startButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
        _startButton.titleLabel.font = [UIFont systemFontOfSize:20];
        
        [self.view addSubview:_startButton];
        
        [_startButton addTarget:self action:@selector(p_tapStartButton) forControlEvents:UIControlEventTouchUpInside];
    }
    return _startButton;
}

@end

File: GYTableViewController4Swift/GYTableViewController4Swift/Demo/CreateTableDemo/CreateTableViewController.swift
//
//  CreateTableViewController.swift
//  GYTableViewController4Swift
//
//  Created by gaoyang on 2019/2/7.
//  Copyright © 2019年 高扬. All rights reserved.
//

import UIKit

class CreateTableViewController: UIViewController {

    //MARK: - 主角是PopupListView中创建的自定义TableView实例
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.backgroundColor = UIColor.white
        self.startButton.frame = CGRect.init(x: 0, y: 0, width: self.view.width, height: 50)
        
        self.popupListView.show()
    }
    
    @objc func p_tapStartButton() {
        self.popupListView.show()
    }
    
    //MARK: - 懒加载添加视图
    private lazy var popupListView:PopupListView = { () -> PopupListView in
        let _popupListView = PopupListView()
        _popupListView.title = "世界上最牛B的语言"
        _popupListView.dataArray = Mock.popupOptions
        return _popupListView
    }()
    
    private lazy var startButton:UIButton = { () -> UIButton in
        let _startButton = UIButton.init(type: UIButton.ButtonType.system)
        _startButton.backgroundColor = TVStyle.colorPrimaryDishes
        _startButton.setTitle("弹  出", for: UIControl.State.normal)
        _startButton.setTitleColor(UIColor.white, for: UIControl.State.normal)
        _startButton.titleLabel?.font = UIFont.systemFont(ofSize: 20)
        self.view.addSubview(_startButton)
        
        _startButton.addTarget(self, action: #selector(p_tapStartButton), for: UIControl.Event.touchUpInside)
        return _startButton
    }()
    
}
